go/api/transaction_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+import (
+	"fmt"
+)
+
+func ExampleTransaction_Exec() {
+	var tx *Transaction = getExampleTransactionGlideClient() // example helper function
+	cmd := tx.GlideClient
+	tx.Discard()
+	cmd.Set("key123", "Glide")
+	cmd.Set("key1", "Glide")
+	cmd.Set("key2", "Hello")
+	cmd.Set("key3", "KeyToDelete")
+	cmd.Del([]string{"key3"})
+	cmd.Append("key2", "_World")
+	cmd.Get("key2")
+	cmd.Set("key123", "Valkey")
+	cmd.Get("key123")
+	cmd.Type("key123")
+	result, _ := tx.Exec()
+	fmt.Println(result)
+
+	// Output:
+	// [OK OK OK OK 1 11 Hello_World OK Valkey string]
+}
+
+func ExampleTransaction_Watch() {
+	var clientTx *Transaction = getExampleTransactionGlideClient() // example helper function
+	cmd := clientTx.GlideClient
+	clientTx.Discard()
+	cmd.Set("key123", "Glide")
+	cmd.Watch([]string{"key123", "key345"})
+	cmd.Get("key123")
+	cmd.Del([]string{"key123"})
+
+	result, _ := clientTx.Exec()
```

Comment:
It would be great to see the recommended pattern for checking that the transaction failed due to the watch being violated and how callers are supposed to detect that and retry (vs. for example syntactical issues with the queued commands or semantic problems of using the wrong family of commands for a different type of key).